core/src/main/java/org/ehcache/core/spi/store/Store.java

```diff
@@ -140,7 +143,7 @@
    *
    * @see #replace(Object, Object)
    */
-  ValueHolder<V> putIfAbsent(K key, V value) throws StoreAccessException;
+  ValueHolder<V> putIfAbsent(K key, V value, Consumer<Boolean> put) throws StoreAccessException;
```

Comment:
I'm thinking/hoping the `Consumer` can be removed.

Reply:
The same thought came tom my mind for the very first time, but after discussing with Henri and he spending trying to get rid of it, following is what is going on
https://github.com/ehcache/ehcache3/pull/2467/commits/bb7f52208e3143c9a2c2b70aa12bfa10706ec2e5

Reply:
The same thought came tom my mind for the very first time, but after discussing with Henri and he spending trying to get rid of it, following is what is going on
`https://github.com/ehcache/ehcache3/pull/2467/commits/bb7f52208e3143c9a2c2b70aa12bfa10706ec2e5`